src/profile/context/ProfileContext.tsx

```diff
@@ -0,0 +1,173 @@
+import { ApolloError } from '@apollo/client';
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import { ProfileRoot } from '../../graphql/typings';
+import getVerifiedPersonalInformation from '../helpers/getVerifiedPersonalInformation';
+import { useProfileQuery, QueryResult } from '../hooks/useProfileQuery';
+
+type ContextProps = {
+  children: React.ReactNode | React.ReactNode[] | null;
+};
+
+type ErrorListener = (e: ApolloError | Error) => void;
+type ErrorDisposer = () => void;
+
+export type ProfileContextData = {
+  data?: ProfileRoot;
+  updateProfileData: (data: ProfileRoot) => void;
+  error?: ApolloError | Error;
+  loading: boolean;
+  addErrorListener: (listener: ErrorListener) => ErrorDisposer;
+  fetch: () => void;
+  refetch: () => Promise<QueryResult>;
+  isInitialized: boolean;
+  isComplete: boolean;
+  getName: (preferNickOrGivenName?: boolean) => string;
```

Comment:
Sonar also marks these boolean "double definitions" as code smells, but in case of an arrow function, adding a default value will result in compiler error.

This is ok:

const getName1 = (preferNickOrGivenName = false):string=>{return preferNickOrGivenName ? 'a':'b'}; 


Sadly, this is not:

export type ProfileContextData = {
  getName: (preferNickOrGivenName = false) => string;
};

Error: A parameter initializer is only allowed in a function or constructor implementation.
Same happens in latest TS version in https://www.typescriptlang.org/play?ts=4.3.0-beta#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgBSgnQQBtgBhCFUGAEUxkzjgG8AoFhAZwEkkEMBJlIIAXsAAmALjgAjCBHKYkAbk6JuVALZhyqWQqXAV6lgHNgMAHKZtwWQAowUYOjTWE2ANYB5KADiCABuwEi29gC88orKSACUcJEAfHDcMFDI5uoAvuqgkLBwMACeYHiExGSU1Kgg9IzM0RwWVhEOcM6u7lCePv5BoeF2eNFYpNzAiSlpGVlmcJY2IwBMTi5uHl5+gSFh7QD80kZx06npmUjZ7Hns2NTpi20jAIxJnRs9fTuD+yPv40m8WkFyyKVYrhgAFcoEg4J8tv1dkN2nADnAAOSYDHSDFyDF5OB3B7wJbtFbvLqbXrbAZ7Yb2I4xYwqYGgq7gyEwuEImlI34MvDorE4vEE1RE9joKFIbBCahPZb2ADMVK+tORf3sAJEQJBcyuEKs3Ph3URP3pqOF2Nx+MJ7GJSEeTUV7Rejni6nuTvgmAp0TJr0cGShUy9JLgmGV70D9ndgLDQA

Reply:
Right. No default values in type definitions. Makes sense. So in the type, you have to mark the argument as optional. Then in the functions implementing that type you can use default values for (optional) arguments. So this is possible:

```typescript
type MyType = {
  f: (flag?: boolean) => string;
};

const myValue: MyType = {
  f: (flag: boolean = false): string => {
    return typeof flag + String(flag);
  }
};

console.log(myValue.f());
console.log(myValue.f(true));
console.log(myValue.f(false));
```

That way the argument is always boolean within the function. I don't know what Sonar says about this combination.

Reply:
The function itself does not need to have ":boolean" again.

const myValue: MyType = {
  f: (flag = false): string => {
    return typeof flag + String(flag);
  }
};

is valid. 

Added this to the file.